lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2197,12 +2197,15 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
       LookUpConformanceInModule(D->getDeclContext()->getParentModule());
 
   FuncDecl *original = nullptr;
-  if (isa<VarDecl>(D)) {
+  bool isProperty = false;
+  if (auto *vd = dyn_cast<VarDecl>(D)) {
     // When used on a storage decl, @differentiable refers to its getter.
-    original = cast<VarDecl>(D)->getGetter();
-  } else if (isa<FuncDecl>(D)) {
-    original = cast<FuncDecl>(D);
+    original = vd->getGetter();
+    isProperty = true;
+  } else if (auto *fd = dyn_cast<FuncDecl>(D)) {
+    original = fd;
```

Comment:
In addition to normal func decls, I believe this case is also triggered by
```
var foo: Float {
  @differentiable
  get { ... }
}
```

So if you want this logic to also apply to that case, you should add something that distinguishes between normal func decls and getter func decls.

Perhaps this would work:
```
bool isProperty = isa<AccessorDecl>(original);
```